DAQ_System/Source/L3GD20H.h

```diff
@@ -0,0 +1,90 @@
+/* Copyright (c) 2011 Pololu Corporation.  For more information, see
+ * 
+ * http://www.pololu.com/
+ * http://forum.pololu.com/
+ * 
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following
+ * conditions:
+ * 
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef __L3GD20H_H
+#define __L3GD20H_H
+
+#include "mbed.h"
+
+#define L3GD20_WHO_AM_I      0x0F
+
+#define L3GD20_CTRL_REG1     0x20
+#define L3GD20_CTRL_REG2     0x21
+#define L3GD20_CTRL_REG3     0x22
+#define L3GD20_CTRL_REG4     0x23
+#define L3GD20_CTRL_REG5     0x24
+#define L3GD20_REFERENCE     0x25
+#define L3GD20_OUT_TEMP      0x26
+#define L3GD20_STATUS_REG    0x27
+
+#define L3GD20_OUT_X_L       0x28
+#define L3GD20_OUT_X_H       0x29
+#define L3GD20_OUT_Y_L       0x2A
+#define L3GD20_OUT_Y_H       0x2B
+#define L3GD20_OUT_Z_L       0x2C
+#define L3GD20_OUT_Z_H       0x2D
+
+#define L3GD20_FIFO_CTRL_REG 0x2E
+#define L3GD20_FIFO_SRC_REG  0x2F
+
+#define L3GD20_INT1_CFG      0x30
+#define L3GD20_INT1_SRC      0x31
+#define L3GD20_INT1_THS_XH   0x32
+#define L3GD20_INT1_THS_XL   0x33
+#define L3GD20_INT1_THS_YH   0x34
+#define L3GD20_INT1_THS_YL   0x35
+#define L3GD20_INT1_THS_ZH   0x36
+#define L3GD20_INT1_THS_ZL   0x37
+#define L3GD20_INT1_DURATION 0x38
+
+
+class L3GD20H{
```

Comment:
Why not move the content of this class to the new `Gyroscope_L3GD20H` class?

Reply:
we could move it to Gyroscope_L3GD20H. I just didn't do that since it didn't seem necessary/ I didn't get errors for not moving it. It seems I can access all of these #define and stuff without having to directly move the contents to the new Gyroscope_L3GD20H class. In addition, the library barely utilizes any of these definitions and it made the abstract interface more sleek as well(this is totally just my opinion). They just seem to be there for the sake of completeness. All the registers and the address(in hex) can be seen below.

<img width="605" alt="Screenshot 2023-06-19 212316" src="https://github.com/DallasFormulaRacing/DataAcquisition2.0/assets/113472670/533262c7-3274-4cd0-8e62-b313b0f8e1c0">

I will note that the only reason why I added #define GYR_ADDRESS 0xD6 to the new class was that this #define was included in the L3GD20H.cpp which is no longer accessible as I don't #include it in any of the files or something like that I'm not really sure why.

with all of that said, moving all of this content to the new class could be a good idea for the readability of the code. Especially if we decide to expand on the gyroscope code/functionality and start using more of these registers. Then having them there in the new class could be really helpful for people who are reading the code for the first time. If you want me to move these contents to the new Gyroscope_L3GD20H class then I can do that.

Reply:
I see what you mean. Though I think it's best to consolidate it under a single class to reduce as much code necessary to support the gyroscope. The less code there is, the less there is to maintain/read and less flash memory will be used.

In some cases, a "wrapper class" (which is what you have here) can be useful to have, but I think it is not necessary here.

If possible, let's go ahead and move the contents  to your new class.

Reply:
OK, I will do that